proxy/reverse.go

```diff
@@ -203,3 +210,39 @@ func maybeSetForwardedFor(req *http.Request) {
 	}
 	req.Header.Set("X-Forwarded-For", clientIP)
 }
+
+// checkProxyLoop returns an error when it detects a proxy loop.
+func checkProxyLoop(req *http.Request, proxyID string, hopLimit int) error {
+	// check if proxyID has been seen before
+	seen := false
+	if v, ok := req.Header["X-Etcd-Proxy-Hop-Seen-By"]; ok {
+		for _, id := range strings.Split(strings.Join(v, ", "), ", ") {
+			if id == proxyID {
+				seen = true
+				break
+			}
+		}
+		// fold multiple proxyIDs into one header
+		proxyID = strings.Join(v, ", ") + ", " + proxyID
+	}
+	req.Header.Set("X-Etcd-Proxy-Hop-Seen-By", proxyID)
```

Comment:
Do you have any plan about how to get the endpoint that needs to be forbidden in the follow-up? Could we use as few headers as possible to make it simple?